File: src/core/constants/appConstants.ts
/**
 * Все «магические числа» из ТЗ собраны в одном месте.
 * Менять пороги, таймауты и лимиты — только здесь.
 */

// ---------- Confidence thresholds ----------

/** Минимальный confidence для озвучивания обычного объекта */
export const CONFIDENCE_THRESHOLD_DEFAULT = 0.45;

/** Пониженный порог для объектов безопасности (ступени, traffic light, stop sign) */
export const CONFIDENCE_THRESHOLD_SAFETY = 0.35;

/** Минимальный confidence для классификации купюр */
export const CONFIDENCE_THRESHOLD_BANKNOTE = 0.6;

/** Минимальный confidence OCR-блока для озвучивания */
export const CONFIDENCE_THRESHOLD_OCR = 0.4;

/** IoU-порог для Non-Maximum Suppression */
export const NMS_IOU_THRESHOLD = 0.45;

// ---------- Timeouts (ms) ----------

/** Авто-отключение камеры после бездействия */
export const CAMERA_INACTIVITY_TIMEOUT_MS = 30_000;

/** Время ожидания голосовой команды (ASR) */
export const ASR_LISTEN_TIMEOUT_MS = 5_000;

/** Время бездействия модели в фоне до выгрузки */
export const MODEL_DISPOSE_TIMEOUT_MS = 60_000;

/** Таймаут для промежуточного сообщения «Обрабатываю…» */
export const PROCESSING_FEEDBACK_DELAY_MS = 1_500;

// ---------- Image sizes ----------

/** Входной размер SSD MobileNet v2 */
export const SSD_INPUT_SIZE = 300;

/** Входной размер MobileNet v3-Small (купюры) */
export const BANKNOTE_INPUT_SIZE = 224;

/** Максимальное разрешение захвата камеры */
export const CAMERA_MAX_WIDTH = 1280;
export const CAMERA_MAX_HEIGHT = 960;

/** Максимальный размер изображения (downscale если больше) */
export const IMAGE_MAX_DIMENSION = 4096;

/** Crop-доля для купюр (центральная область) */
export const BANKNOTE_CROP_RATIO = 0.6;

// ---------- Limits ----------

/** Максимум записей в истории распознаваний */
export const HISTORY_MAX_ENTRIES = 50;

/** Максимальный размер лог-файла (байт) */
export const LOG_MAX_FILE_SIZE_BYTES = 5 * 1024 * 1024; // 5 MB

/** Количество хранимых лог-файлов */
export const LOG_MAX_FILES = 3;

/** Количество объектов для озвучивания (топ-N) */
export const DETECTION_TOP_N = 3;

// ---------- TTS ----------

/** Минимальная скорость речи */
export const TTS_RATE_MIN = 0.5;

/** Максимальная скорость речи */
export const TTS_RATE_MAX = 2.0;

/** Скорость речи по умолчанию */
export const TTS_RATE_DEFAULT = 1.0;

// ---------- UI ----------

/** Минимальный touch-target (dp) для accessibility */
export const MIN_TOUCH_TARGET = 48;

File: lib/core/constants/app_constants.dart
/// Глобальные константы приложения assist-eye.
///
/// Все «магические числа» из ТЗ собраны здесь, чтобы менять
/// поведение в одном месте, а не искать по всему коду.
class AppConstants {
  AppConstants._(); // Запрет создания экземпляров

  // --------------- Камера ---------------
  /// Максимальное разрешение кадра для OCR (ширина × высота).
  /// Баланс качества и скорости (§5.1 ТЗ).
  static const int cameraMaxWidth = 1280;
  static const int cameraMaxHeight = 960;

  /// Таймаут бездействия камеры — после этого камера авто-деактивируется (§4.6).
  static const Duration cameraInactivityTimeout = Duration(seconds: 30);

  // --------------- ASR ---------------
  /// Таймаут записи голоса (§5.1 ТЗ: ≤ 5 с).
  static const Duration asrListenTimeout = Duration(seconds: 5);

  // --------------- OCR ---------------
  /// Минимальный confidence для блока текста (§5.1e).
  static const double ocrMinConfidence = 0.4;

  /// Максимальный размер изображения — downscale если больше (§5.3).
  static const int imageMaxDimension = 4096;

  // --------------- Детекция объектов ---------------
  /// Размер входного тензора SSD MobileNet v2 (§5.1 сценарий «Опиши»).
  static const int detectorInputSize = 300;

  /// IoU-порог для Non-Maximum Suppression (§ТЗ: 0.45).
  static const double nmsIouThreshold = 0.45;

  /// Минимальный confidence для обычных объектов (§4.3).
  static const double detectionConfidenceThreshold = 0.45;

  /// Повышенный confidence-порог для объектов безопасности
  /// (ступенька, traffic light, stop sign — Приложение A).
  static const double safetyConfidenceThreshold = 0.35;

  /// Максимальное количество озвучиваемых объектов (§5.1: топ-3).
  static const int maxDetectionsToAnnounce = 3;

  // --------------- Классификатор купюр ---------------
  /// Размер входного тензора MobileNet v3-Small (§5.1 сценарий «Купюра»).
  static const int banknoteInputSize = 224;

  /// Crop центральной области кадра (§5.1: 60%).
  static const double banknoteCropRatio = 0.6;

  /// Минимальный confidence для купюры (§5.1: 0.6).
  static const double banknoteMinConfidence = 0.6;

  // --------------- Хранилище ---------------
  /// Максимальное число записей в истории (§4.5).
  static const int historyMaxEntries = 50;

  /// Имя файла БД SQLite.
  static const String databaseName = 'assist_eye.db';

  /// Версия схемы БД (для миграций).
  static const int databaseVersion = 1;

  // --------------- Логирование ---------------
  /// Максимальный размер лог-файла перед ротацией (§8.2).
  static const int logMaxFileSizeBytes = 5 * 1024 * 1024; // 5 MB

  /// Количество файлов лога для хранения.
  static const int logMaxFileCount = 3;

  // --------------- ML-модели: жизненный цикл ---------------
  /// Таймаут перед выгрузкой моделей при уходе в фон (§8.3: >60 с).
  static const Duration modelDisposeDelay = Duration(seconds: 60);

  // --------------- UX: обратная связь ---------------
  /// Порог длительной операции для показа «Обрабатываю…» (§5.2: >1.5 с).
  static const Duration longOperationThreshold = Duration(milliseconds: 1500);
}
